### ISSUE TITLE
Crash when attempting to drag window statusline/separator in another tabpage

### ISSUE THREAD

**Issue by zeertzjq** [vim/vim#11427]
### Steps to reproduce

Crash 1:
```vim
let id = win_getid()
tabnew
call win_move_statusline(id, -1)
```

Crash 2:
```vim
set laststatus=2
set mouse=a
func ClickExpr()
  call test_setmouse(&lines - 1, 1)
  return "\<LeftMouse>"
endfunc
func DragExpr()
  call test_setmouse(&lines - 2, 1)
  return "\<LeftDrag>"
endfunc
nnoremap <expr> <F2> ClickExpr()
nnoremap <expr> <F3> DragExpr()
call feedkeys("\<F2>:tabnew\<CR>\<F3>", 'tx')
```

Crash 3:
```vim
let id = win_getid()
tabnew
call win_move_separator(id, -1)
```

### Expected behaviour

No crash in any of these cases

### Version of Vim

9.0.815

### Environment

Operating system: Arch Linux
Terminal: kitty
Value of $TERM: xterm-kitty
Shell: fish

**Comment by zeertzjq** [comment]
Crash 1:
```
window.c:6104:10: runtime error: member access within null pointer of type 'frame_T' (aka 'struct frame_S')
    #0 0x55b8ef63f757 in win_drag_status_line **/src/window.c:6104:10
    #1 0x55b8ee62d44b in f_win_move_statusline **/src/evalwindow.c:919:5
    #2 0x55b8ee5793f2 in call_internal_func **/src/evalfunc.c:3050:5
    #3 0x55b8ef3c9511 in call_func **/src/userfunc.c:3681:14
    #4 0x55b8ef3c4756 in get_func_tv **/src/userfunc.c:1841:8
    #5 0x55b8ef40d6c9 in ex_call_inner **/src/userfunc.c:5647:6
    #6 0x55b8ef40b4c5 in ex_call **/src/userfunc.c:5971:11
    #7 0x55b8ee697e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #8 0x55b8ee683f99 in do_cmdline **/src/ex_docmd.c:990:17
    #9 0x55b8eef64612 in do_source_ext **/src/scriptfile.c:1667:5
    #10 0x55b8eef5f848 in cmd_source **/src/scriptfile.c:1146:6
    #11 0x55b8eef5f2e9 in ex_source **/src/scriptfile.c:1189:2
    #12 0x55b8ee697e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #13 0x55b8ee683f99 in do_cmdline **/src/ex_docmd.c:990:17
    #14 0x55b8eeb73018 in nv_colon **/src/normal.c:3183:19
    #15 0x55b8eeb3c8a4 in normal_cmd **/src/normal.c:920:5
    #16 0x55b8ef8d466d in main_loop **/src/main.c:1524:3
    #17 0x55b8ef8cdea9 in vim_main2 **/src/main.c:886:5
    #18 0x55b8ef8bdd96 in main **/src/main.c:432:12
    #19 0x7f93412c528f  (/usr/lib/libc.so.6+0x2328f) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #20 0x7f93412c5349 in __libc_start_main (/usr/lib/libc.so.6+0x23349) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #21 0x55b8ee144934 in _start /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior window.c:6104:10 in 
window.c:6104:10: runtime error: load of null pointer of type 'char'
    #0 0x55b8ef63f787 in win_drag_status_line **/src/window.c:6104:10
    #1 0x55b8ee62d44b in f_win_move_statusline **/src/evalwindow.c:919:5
    #2 0x55b8ee5793f2 in call_internal_func **/src/evalfunc.c:3050:5
    #3 0x55b8ef3c9511 in call_func **/src/userfunc.c:3681:14
    #4 0x55b8ef3c4756 in get_func_tv **/src/userfunc.c:1841:8
    #5 0x55b8ef40d6c9 in ex_call_inner **/src/userfunc.c:5647:6
    #6 0x55b8ef40b4c5 in ex_call **/src/userfunc.c:5971:11
    #7 0x55b8ee697e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #8 0x55b8ee683f99 in do_cmdline **/src/ex_docmd.c:990:17
    #9 0x55b8eef64612 in do_source_ext **/src/scriptfile.c:1667:5
    #10 0x55b8eef5f848 in cmd_source **/src/scriptfile.c:1146:6
    #11 0x55b8eef5f2e9 in ex_source **/src/scriptfile.c:1189:2
    #12 0x55b8ee697e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #13 0x55b8ee683f99 in do_cmdline **/src/ex_docmd.c:990:17
    #14 0x55b8eeb73018 in nv_colon **/src/normal.c:3183:19
    #15 0x55b8eeb3c8a4 in normal_cmd **/src/normal.c:920:5
    #16 0x55b8ef8d466d in main_loop **/src/main.c:1524:3
    #17 0x55b8ef8cdea9 in vim_main2 **/src/main.c:886:5
    #18 0x55b8ef8bdd96 in main **/src/main.c:432:12
    #19 0x7f93412c528f  (/usr/lib/libc.so.6+0x2328f) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #20 0x7f93412c5349 in __libc_start_main (/usr/lib/libc.so.6+0x23349) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #21 0x55b8ee144934 in _start /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior window.c:6104:10 in 
```

Crash 2:
```
window.c:6104:10: runtime error: member access within null pointer of type 'frame_T' (aka 'struct frame_S')
    #0 0x562ccb38d757 in win_drag_status_line **/src/window.c:6104:10
    #1 0x562cca824389 in jump_to_mouse **/src/mouse.c:1821:6
    #2 0x562cca818e05 in do_mouse **/src/mouse.c:710:18
    #3 0x562cca8343d0 in nv_mouse **/src/mouse.c:2166:11
    #4 0x562cca88a8a4 in normal_cmd **/src/normal.c:920:5
    #5 0x562cca43045f in exec_normal **/src/ex_docmd.c:8853:6
    #6 0x562cca2e04e3 in f_feedkeys **/src/evalfunc.c:4410:3
    #7 0x562cca2c73f2 in call_internal_func **/src/evalfunc.c:3050:5
    #8 0x562ccb117511 in call_func **/src/userfunc.c:3681:14
    #9 0x562ccb112756 in get_func_tv **/src/userfunc.c:1841:8
    #10 0x562ccb15b6c9 in ex_call_inner **/src/userfunc.c:5647:6
    #11 0x562ccb1594c5 in ex_call **/src/userfunc.c:5971:11
    #12 0x562cca3e5e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #13 0x562cca3d1f99 in do_cmdline **/src/ex_docmd.c:990:17
    #14 0x562ccacb2612 in do_source_ext **/src/scriptfile.c:1667:5
    #15 0x562ccacad848 in cmd_source **/src/scriptfile.c:1146:6
    #16 0x562ccacad2e9 in ex_source **/src/scriptfile.c:1189:2
    #17 0x562cca3e5e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #18 0x562cca3d1f99 in do_cmdline **/src/ex_docmd.c:990:17
    #19 0x562cca8c1018 in nv_colon **/src/normal.c:3183:19
    #20 0x562cca88a8a4 in normal_cmd **/src/normal.c:920:5
    #21 0x562ccb62266d in main_loop **/src/main.c:1524:3
    #22 0x562ccb61bea9 in vim_main2 **/src/main.c:886:5
    #23 0x562ccb60bd96 in main **/src/main.c:432:12
    #24 0x7fd57688628f  (/usr/lib/libc.so.6+0x2328f) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #25 0x7fd576886349 in __libc_start_main (/usr/lib/libc.so.6+0x23349) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #26 0x562cc9e92934 in _start /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior window.c:6104:10 in 
window.c:6104:10: runtime error: load of null pointer of type 'char'
    #0 0x562ccb38d787 in win_drag_status_line **/src/window.c:6104:10
    #1 0x562cca824389 in jump_to_mouse **/src/mouse.c:1821:6
    #2 0x562cca818e05 in do_mouse **/src/mouse.c:710:18
    #3 0x562cca8343d0 in nv_mouse **/src/mouse.c:2166:11
    #4 0x562cca88a8a4 in normal_cmd **/src/normal.c:920:5
    #5 0x562cca43045f in exec_normal **/src/ex_docmd.c:8853:6
    #6 0x562cca2e04e3 in f_feedkeys **/src/evalfunc.c:4410:3
    #7 0x562cca2c73f2 in call_internal_func **/src/evalfunc.c:3050:5
    #8 0x562ccb117511 in call_func **/src/userfunc.c:3681:14
    #9 0x562ccb112756 in get_func_tv **/src/userfunc.c:1841:8
    #10 0x562ccb15b6c9 in ex_call_inner **/src/userfunc.c:5647:6
    #11 0x562ccb1594c5 in ex_call **/src/userfunc.c:5971:11
    #12 0x562cca3e5e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #13 0x562cca3d1f99 in do_cmdline **/src/ex_docmd.c:990:17
    #14 0x562ccacb2612 in do_source_ext **/src/scriptfile.c:1667:5
    #15 0x562ccacad848 in cmd_source **/src/scriptfile.c:1146:6
    #16 0x562ccacad2e9 in ex_source **/src/scriptfile.c:1189:2
    #17 0x562cca3e5e35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #18 0x562cca3d1f99 in do_cmdline **/src/ex_docmd.c:990:17
    #19 0x562cca8c1018 in nv_colon **/src/normal.c:3183:19
    #20 0x562cca88a8a4 in normal_cmd **/src/normal.c:920:5
    #21 0x562ccb62266d in main_loop **/src/main.c:1524:3
    #22 0x562ccb61bea9 in vim_main2 **/src/main.c:886:5
    #23 0x562ccb60bd96 in main **/src/main.c:432:12
    #24 0x7fd57688628f  (/usr/lib/libc.so.6+0x2328f) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #25 0x7fd576886349 in __libc_start_main (/usr/lib/libc.so.6+0x23349) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #26 0x562cc9e92934 in _start /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior window.c:6104:10 in 
```

Crash 3:
```
window.c:6233:13: runtime error: member access within null pointer of type 'frame_T' (aka 'struct frame_S')
    #0 0x557a33695493 in win_drag_vsep_line **/src/window.c:6233:13
    #1 0x557a3268116b in f_win_move_separator **/src/evalwindow.c:894:5
    #2 0x557a325cd3f2 in call_internal_func **/src/evalfunc.c:3050:5
    #3 0x557a3341d511 in call_func **/src/userfunc.c:3681:14
    #4 0x557a33418756 in get_func_tv **/src/userfunc.c:1841:8
    #5 0x557a334616c9 in ex_call_inner **/src/userfunc.c:5647:6
    #6 0x557a3345f4c5 in ex_call **/src/userfunc.c:5971:11
    #7 0x557a326ebe35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #8 0x557a326d7f99 in do_cmdline **/src/ex_docmd.c:990:17
    #9 0x557a32fb8612 in do_source_ext **/src/scriptfile.c:1667:5
    #10 0x557a32fb3848 in cmd_source **/src/scriptfile.c:1146:6
    #11 0x557a32fb32e9 in ex_source **/src/scriptfile.c:1189:2
    #12 0x557a326ebe35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #13 0x557a326d7f99 in do_cmdline **/src/ex_docmd.c:990:17
    #14 0x557a32bc7018 in nv_colon **/src/normal.c:3183:19
    #15 0x557a32b908a4 in normal_cmd **/src/normal.c:920:5
    #16 0x557a3392866d in main_loop **/src/main.c:1524:3
    #17 0x557a33921ea9 in vim_main2 **/src/main.c:886:5
    #18 0x557a33911d96 in main **/src/main.c:432:12
    #19 0x7faf55e3528f  (/usr/lib/libc.so.6+0x2328f) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #20 0x7faf55e35349 in __libc_start_main (/usr/lib/libc.so.6+0x23349) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #21 0x557a32198934 in _start /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior window.c:6233:13 in 
window.c:6233:13: runtime error: load of null pointer of type 'char'
    #0 0x557a336954c3 in win_drag_vsep_line **/src/window.c:6233:13
    #1 0x557a3268116b in f_win_move_separator **/src/evalwindow.c:894:5
    #2 0x557a325cd3f2 in call_internal_func **/src/evalfunc.c:3050:5
    #3 0x557a3341d511 in call_func **/src/userfunc.c:3681:14
    #4 0x557a33418756 in get_func_tv **/src/userfunc.c:1841:8
    #5 0x557a334616c9 in ex_call_inner **/src/userfunc.c:5647:6
    #6 0x557a3345f4c5 in ex_call **/src/userfunc.c:5971:11
    #7 0x557a326ebe35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #8 0x557a326d7f99 in do_cmdline **/src/ex_docmd.c:990:17
    #9 0x557a32fb8612 in do_source_ext **/src/scriptfile.c:1667:5
    #10 0x557a32fb3848 in cmd_source **/src/scriptfile.c:1146:6
    #11 0x557a32fb32e9 in ex_source **/src/scriptfile.c:1189:2
    #12 0x557a326ebe35 in do_one_cmd **/src/ex_docmd.c:2578:2
    #13 0x557a326d7f99 in do_cmdline **/src/ex_docmd.c:990:17
    #14 0x557a32bc7018 in nv_colon **/src/normal.c:3183:19
    #15 0x557a32b908a4 in normal_cmd **/src/normal.c:920:5
    #16 0x557a3392866d in main_loop **/src/main.c:1524:3
    #17 0x557a33921ea9 in vim_main2 **/src/main.c:886:5
    #18 0x557a33911d96 in main **/src/main.c:432:12
    #19 0x7faf55e3528f  (/usr/lib/libc.so.6+0x2328f) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #20 0x7faf55e35349 in __libc_start_main (/usr/lib/libc.so.6+0x23349) (BuildId: 1e94beb079e278ac4f2c8bce1f53091548ea1584)
    #21 0x557a32198934 in _start /build/glibc/src/glibc/csu/../sysdeps/x86_64/start.S:115

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior window.c:6233:13 in 
```

**Comment by zeertzjq** [comment]
The first two crashes have been fixed by patch 9.0.0821 and 9.0.0822. I've created PR #11479 to fix the third crash.

**Comment by zeertzjq** [comment]
I've also created a PR #11483 to fix a regression introduced by the fix for the second crash.
